Documentation/policy/troubleshooting.rst

```diff
@@ -175,95 +176,101 @@ Troubleshooting ``toFQDNs`` rules
-                        "cilium-generated:ToFQDN-UUID=9a1d4006-1360-11e9-a6d4-080027d2d952"
-                      ]
-                    ],
-                    "rule": "104.198.14.52/32"
+                    "port": "53",
+                    "protocol": "ANY"
                   }
                 ],
-                "ingress": []
+                "rules": {
+                  "dns": [
+                    {
+                      "matchPattern": "*"
+                    }
+                  ]
+                }
               }
-            }
+            ]
+          }
+        ],
+        "labels": [
+          {
+            "key": "io.cilium.k8s.policy.derived-from",
+            "value": "CiliumNetworkPolicy",
+            "source": "k8s"
+          },
+          {
+            "key": "io.cilium.k8s.policy.name",
+            "value": "fqdn",
+            "source": "k8s"
+          },
+          {
+            "key": "io.cilium.k8s.policy.namespace",
+            "value": "default",
+            "source": "k8s"
+          },
+          {
+            "key": "io.cilium.k8s.policy.uid",
+            "value": "fc9d6022-2ffa-4f72-b59e-b9067c3cfecf",
+            "source": "k8s"
+          }
+        ]
+      }
 
-.. only:: epub or latex
 
-   .. code-block:: json
+#. After making a DNS request, the FQDN to IP mapping should be available via
+   ``cilium fqdn cache list``:
 
-      {
-        "cidr-policy": {
-          "egress": [
-            {
-              "derived-from-rules": [
-                [
-                  "k8s:io.cilium.k8s.policy.name=rebel-escape",
-                  "k8s:io.cilium.k8s.policy.uid=c96f66a8-135e-11e9-babd-080027d2d952",
-                  "k8s:io.cilium.k8s.policy.namespace=default",
-                  "k8s:io.cilium.k8s.policy.derived-from=CiliumNetworkPolicy",
-                  "cilium-generated:ToFQDN-UUID=9a1d4006-1360-11e9-a6d4-080027d2d952"
-                ]
-              ],
-              "rule": "104.198.14.52/32"
-            }
-          ],
-          "ingress": []
-        }
-      }
+   .. code-block:: shell-session
+
+      # cilium fqdn cache list
+      Endpoint   FQDN                TTL      ExpirationTime             IPs
+      2761       help.twitter.com.   604800   2019-07-16T17:57:38.179Z   104.244.42.67,104.244.42.195,104.244.42.3,104.244.42.131
+      2761       api.twitter.com.    604800   2019-07-16T18:11:38.627Z   104.244.42.194,104.244.42.130,104.244.42.66,104.244.42.2
+
+#. If the traffic is allowed, then these IPs should have corresponding local identities via
+   ``cilium identity list | grep <IP>``:
+
+   .. code-block:: shell-session
+
+      # cilium identity list | grep -A 1 104.244.42.194
+      16777220   cidr:104.244.42.194/32
+                 reserved:world
 
+#. Given the identity of the traffic that should be allowed, the regular
+   :ref:`policy_tracing` steps can be used to validate that the policy is
```

Comment:
This isn't the case for egress, though :/

Reply:
True, but I don't think it's worth documenting that limitation here.